pyomo/contrib/solver/solvers/sol_reader.py

```diff
@@ -122,6 +122,12 @@ def get_duals(
                 'Solution loader does not currently have a valid solution. Please '
                 'check results.termination_condition and/or results.solution_status.'
             )
+        # If the NL instance has no objectives, report zeros
+        if len(self._nl_info.objectives) == 0:
```

Comment:
Style: I would generally prefer (and it is measurably faster):

```suggestion
        if not self._nl_info.objectives:
```